libsql-server/src/http/user/listen.rs

```diff
@@ -0,0 +1,122 @@
+use std::io::Write;
+
+use axum::extract::{Query, State as AxumState};
+use hyper::HeaderMap;
+use serde::ser::{Serialize, SerializeStruct, Serializer};
+use serde::Deserialize;
+use tokio::io::DuplexStream;
+use tokio::sync::broadcast::error::RecvError;
+use tokio_util::io::{ReaderStream, SyncIoBridge};
+
+use crate::auth::Authenticated;
+use crate::broadcaster::{Action, UpdateHookMsg};
+use crate::error::Error;
+use crate::BLOCKING_RT;
+
+use super::db_factory::namespace_from_headers;
+use super::AppState;
+
+#[derive(Deserialize)]
+pub struct ListenQuery {
+    table: String,
+    action: Option<Action>,
+}
+
+pub(super) async fn handle_listen(
+    auth: Authenticated,
+    AxumState(state): AxumState<AppState>,
+    headers: HeaderMap,
+    query: Query<ListenQuery>,
+) -> crate::Result<axum::body::StreamBody<ReaderStream<DuplexStream>>> {
+    let namespace = namespace_from_headers(
+        &headers,
+        state.disable_default_namespace,
+        state.disable_namespaces,
+    )?;
+
+    if !auth.is_namespace_authorized(&namespace) {
+        return Err(Error::NamespaceDoesntExist(namespace.to_string()));
+    }
+
+    // TODO: validate table
+    let update_broadcaster = state.namespaces.update_broadcaster(namespace).await?;
+    let mut rx = update_broadcaster.subscribe(query.table.as_ref());
+    let action = query.action.clone();
+
+    let (reader, writer) = tokio::io::duplex(8 * 1024);
+
+    BLOCKING_RT.spawn_blocking(|| {
+        futures::executor::block_on(async move {
+            let mut writer = SyncIoBridge::new(writer);
+            // TODO: stop when request is cancelled
+            while recv_listen_msg(&mut writer, &mut rx, &action).await {}
+        })
+    });
```

Comment:
I think all of that would be better handled by: https://docs.rs/axum-extra/0.9.3/axum_extra/json_lines/struct.JsonLines.html

Reply:
all of this could be made very concise with https://docs.rs/tokio-stream/0.1.15/tokio_stream/wrappers/struct.BroadcastStream.html. something like (pseudo rust):

```rust
let stream = BroadcastStream::new(rx).filter(|event| event.action == action).map(|event| SubResponse::from(event));
JsonLines(stream)
```